app/views/application/view_codes.html.erb

```diff
@@ -0,0 +1,45 @@
+<%= render :partial => 'shared/offeror_header' %>
+
+<h4 class="center">
+  Uploaded codes by <%= current_offeror.name %>
+</h4>
+
+<div class="center">
+  <%= link_to image_tag("home2.png", alt: "/#{params[:controller]}/home",
+    size: "40x40"), "/#{params[:controller]}/home" %>
+</div>
+
+<table class="codes_table">
+  <thead>
+    <tr>
+      <th>Code</th>
+      <% if current_vendor %>
+        <th>Value</th>
+        <th>Expiration</th>
+        <th>Description</th>
+        <th>Instruction</th>
+        <th>Help</th>
+      <% end %>
+    </tr>
+  </thead>
+  <tbody>
+    <% @offeror_codes.each do |code| %>
```

Comment:
With no Redemify or vendor codes persisting in the db and related checks missing in the controller action:
```
def view_codes
    @offeror_codes = offeror_codes.all
end
```
the nil set @offeror_codes will lead to a bug,